scripts/charmm_to_fah.py

```diff
@@ -0,0 +1,309 @@
+    max_crds[2] = max(max_crds[2], coord[2])
+
+psf.setBox(max_crds[0]-min_crds[0],
+           max_crds[1]-min_crds[1],
+           max_crds[2]-min_crds[2],
+           )
+
+# Create PDBFixer, retrieving PDB template
+print("creating Modeller...")
+modeller = app.Modeller(pdb.topology, pdb.positions)
+
+# Create directory to store files in.
+workdir = os.path.join(tmp_path, 'RUN'+name)
+if not os.path.exists(workdir):
+    os.makedirs(workdir)
+    print("Creating path %s" % workdir)
+
+# Write PDB file for solute only.
+if verbose: print("Writing initial output...")
+pdb_filename = os.path.join(workdir, 'initial.pdb')
+outfile = open(pdb_filename, 'w')
+app.PDBFile.writeFile(modeller.topology, modeller.positions, outfile)
+outfile.close()
+
+# Create OpenMM system.
+if verbose: print("Creating OpenMM system...")
+#system = psf.createSystem(params, nonbondedMethod=nonbonded_method, nonbondedCutoff=nonbonded_cutoff, constraints=app.HBonds)
+system = psf.createSystem(params, nonbondedMethod=nonbonded_method, nonbondedCutoff=nonbonded_cutoff, constraints=None)
+if verbose: print("Adding barostat...")
+system.addForce(openmm.MonteCarloBarostat(pressure, temperature, barostat_frequency))
+
+# Create simulation.
+if verbose: print("Creating simulation...")
+integrator = openmm.LangevinIntegrator(temperature, collision_rate, timestep)
+#platform = openmm.Platform.getPlatformByName('CPU')
+platform = openmm.Platform.getPlatformByName('OpenCL')
+platform.setPropertyDefaultValue('OpenCLPrecision', 'double') # use double precision
+simulation = app.Simulation(modeller.topology, system, integrator, platform=platform)
+try:
+    simulation.context.setPositions(modeller.positions)
+except Exception as e:
+    print(len(modeller.positions))
+    print(simulation.context.getSystem().getNumParticles())
+    raise(e)
+
+# Minimize energy.
+if verbose: print("Minimizing energy...")
+potential_energy = simulation.context.getState(getEnergy=True).getPotentialEnergy()
+if numpy.isnan(potential_energy / unit.kilocalories_per_mole):
+    raise Exception("Potential energy is NaN before minimization.")
+if verbose: print("Initial potential energy : %10.3f kcal/mol" % (potential_energy / unit.kilocalories_per_mole))
+simulation.minimizeEnergy(maxIterations=max_minimization_iterations)
+potential_energy = simulation.context.getState(getEnergy=True).getPotentialEnergy()
+if numpy.isnan(potential_energy / unit.kilocalories_per_mole):
+    raise Exception("Potential energy is NaN after minimization.")
+if verbose: print("Final potential energy:  : %10.3f kcal/mol" % (potential_energy / unit.kilocalories_per_mole))
+
+del(modeller)
+positions = simulation.context.getState(getPositions=True).getPositions(asNumpy=True)
+modeller = app.Modeller(simulation.topology, positions)
+
+del(system)
+del(integrator)
+del(platform)
+del(simulation.context)
+del(simulation)
+simulation = None
+
+if verbose: print("Creating constrained OpenMM system...")
```

Comment:
Modeller should certainly know the box size at this point. Why do we need to select the min and max coordinates again to set the box size?